native/src/test/scala/sqs4s/internal/aws4/aws4suite/CanonicalRequestSpec.scala

```diff
@@ -0,0 +1,408 @@
+package sqs4s.internal.aws4.aws4suite
+
+import cats.effect.IO
+import org.http4s.Method._
+import org.http4s._
+import org.http4s.client.dsl.io._
+import sqs4s.internal.aws4
+import sqs4s.internal.aws4.IOSpec
+import sqs4s.internal.models.CReq
+
+import scala.io.Source
+
+class CanonicalRequestSpec extends IOSpec {
+
+  import aws4.common._
+
+  def expect(testName: String): String =
+    Source
+      .fromResource(s"aws-sig-v4-test-suite/$testName/$testName.creq")
+      .getLines()
+      .mkString("\n")
+
+  def expectNested(groupName: String, testName: String): String =
+    Source
+      .fromResource(
+        s"aws-sig-v4-test-suite/$groupName/$testName/$testName.creq"
+      )
+      .getLines()
+      .mkString("\n")
+
+  "get header key duplicate" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = uri"/",
+      headers = Headers.of(
+        Header("Host", "example.amazonaws.com"),
+        Header("My-Header1", "value2"),
+        Header("My-Header1", "value2"),
+        Header("My-Header1", "value1"),
+        Header(XAmzDate, testTimeStamp)
+      )
+    )
+
+    CReq[IO](req).value.unsafeRunSync() shouldEqual expect(
+      "get-header-key-duplicate"
+    )
+  }
+
+  "get header value multiline" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = uri"/",
+      headers = Headers.of(
+        Header("Host", "example.amazonaws.com"),
+        Header("My-Header1", """value1
+            |  value2
+            |     value3
+          """.stripMargin),
+        Header(XAmzDate, testTimeStamp)
+      )
+    )
+
+    CReq[IO](req).value.unsafeRunSync() shouldEqual expect(
+      "get-header-value-multiline"
+    )
+  }
+
+  "get unreserved" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri =
+        uri"/-._~0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz",
+      headers = Headers.of(
+        Header("Host", "example.amazonaws.com"),
+        Header(XAmzDate, testTimeStamp)
+      )
+    )
+
+    CReq[IO](req).value.unsafeRunSync() shouldEqual expect("get-unreserved")
+  }
+
+  "get utf8" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = canonicalUri[IO]("/ሴ").unsafeRunSync(),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Non ascii characters are not allowed](https://app.codacy.com/manual/d2a4u/sqs4s/pullRequest?prid=4059007)